planet/src/main/webapp/js/authentication.js

```diff
@@ -22,18 +42,35 @@ function resetForm() {
     }
 }
 
-function submitForm() {
+function signUp() {
     const password = document.getElementById('password').value;
     const passwordConfirmation = document.getElementById('repeat-password').value;
     const email = document.getElementById('email').value;
+    const displayName = document.getElementById('first_name').value 
+    + ' ' + document.getElementById('last_name').value;
+    resetForm('input-sign-up');
     if (password !== passwordConfirmation) {
-        resetForm();
         return;
     }
-    firebase.auth().createUserWithEmailAndPassword(email, password).catch(function(error) {
+    firebase.auth().createUserWithEmailAndPassword(email, password)
+    .then(function() {
+        return firebase.auth().currentUser.updateProfile({
+            displayName: displayName,
+        })
+    }).catch(function(error) {
+        // Handle Errors here.
+        console.log(error);
+        alert('Create account with Email and Password failed!');
```

Comment:
different errors should be handled differently. In particular, if there is an action the user can take to fix the error, we should display a message that tells them what to do.

```suggestion
    }).catch(function(error) {
        console.log(error);
       if (error.code === 'auth/email-already-in-use') {
          alert('There is already an account with this email address. Try signing in instead.');
       } else if (error.code === 'auth/invalid-email') {
         alert('Invalid email address. Double check the spelling.');
       } else if (....) {
        ....
       } else {
            // No specific action the user can take. Show a generic message.
           alert('Create account with Email and Password failed!');
       }
```

Reply:
See https://firebase.google.com/docs/reference/js/firebase.auth.Auth#error-codes_3 and  https://firebase.google.com/docs/reference/js/firebase.auth.Auth#error-codes_12  for the possible error codes

Reply:
Got it! Fixed